vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -805,6 +805,10 @@ void LedgerImpl::DisconnectWallet(const std::string& wallet_type,
   });
 }
 
+void LedgerImpl::DisconnectExternalWallets(ResultCallback callback) {
+  WhenReady([this, callback]() { wallet()->DisconnectAllWallets(callback); });
```

Comment:
This currently doesn't work for the following reasons:

- `DisconnectAllWallets` calls the de-linking endpoints unconditionally, whether the user has an active wallet for a given provider or not.
- `DisconnectAllWallets` calls its callback immediately, without waiting for the de-linking operations to complete.